File: frontend/src/microservices/attraction/objects/AttractionTicketTypeRuleResponse.ts
// 本文件定义 attraction 模块的 `AttractionTicketTypeRuleResponse`，作为响应数据并提供 JSON 编解码。

export type AttractionTicketTypeRuleResponse = {
  ruleId: string
  ruleType: string
  ageValue: number | null
  minAge: number | null
  maxAge: number | null
  documentType: string | null
  documentNumberPrefix: string | null
  summary: string
}
export const attractionTicketTypeRuleResponseFromJson = (json: string): AttractionTicketTypeRuleResponse =>
  JSON.parse(json) as AttractionTicketTypeRuleResponse

export const attractionTicketTypeRuleResponseToJson = (value: AttractionTicketTypeRuleResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/attraction/objects/AttractionTicketTypeRuleResponse.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AttractionTicketTypeRuleResponse(
    ruleId: String,
    ruleType: String,
    ageValue: Option[Int],
    minAge: Option[Int],
    maxAge: Option[Int],
    documentType: Option[String],
    documentNumberPrefix: Option[String],
    summary: String
)
object AttractionTicketTypeRuleResponse:
  given sourceEncoder: Encoder[AttractionTicketTypeRuleResponse] = deriveEncoder
  given sourceDecoder: Decoder[AttractionTicketTypeRuleResponse] = deriveDecoder
